Mathlib/Probability/Variance.lean

```diff
@@ -341,4 +343,60 @@ theorem IndepFun.variance_sum [IsProbabilityMeasure μ] {ι : Type*} {X : ι →
       rw [IH (fun i hi => hs i (mem_insert_of_mem hi))
           (h.mono (by simp only [coe_insert, Set.subset_insert]))]
 
+theorem abs_le_max_abs_abs_ae {a b : ℝ} {X : Ω → ℝ}
+    (ha : ∀ᵐ (ω : Ω) ∂μ, a ≤ X ω) (hb : ∀ᵐ (ω : Ω) ∂μ, X ω ≤ b) :
+    ∀ᵐ (ω : Ω) ∂μ, |X ω| ≤ max |a| |b| := by
+  filter_upwards [ha, hb] with ω using abs_le_max_abs_abs
```

Comment:
Do we really want to turn every deterministic statement ever into a statement about ae?

Reply:
> Do we really want to turn every deterministic statement ever into a statement about ae?

Would it be nice to have a tactic that generates this?

Reply:
It's called `filter_upwards`, as witnessed by the code

Reply:
Yes indeed, but I meant something like `attribute to_ae` that generates `abs_le_max_abs_abs_ae` from `abs_le_max_abs_abs`.

Reply:
Sorry, I was unclear. I mean that we do *not* want to turn every deterministic statement into an ae statement. Instead, we should use `filter_upwards` whenever needed in later proofs. Here this means *not* proving `abs_le_max_ae` but instead calling `filter_upwards` wherever you used `abs_le_max_ae`.